src/app/store/products-store/products.effects.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@angular/core';
+import { Actions, createEffect, ofType } from '@ngrx/effects';
+import { ProductsService } from '../../shared/services/products.service';
+import { Store } from '@ngrx/store';
+import { AppStateInterface } from '../../shared/interfaces/interfaces';
+import * as ProductActions from './products.actions';
+import { map, mergeMap, withLatestFrom } from 'rxjs';
+import { productTypeSelector } from '../product-type-store/product-type.selectors';
+
+@Injectable()
+export class ProductsEffect {
+  loadProducts$ = createEffect(() => {
+    return this.actions.pipe(
+      ofType(ProductActions.addProductsAction),
+      withLatestFrom(this.store.select(productTypeSelector)),
+      mergeMap(([currentPayload, productType]) =>
```

Comment:
Here should be SwitchMap used instead. 
Could you check the difference between mergeMap and switchMap?